libkbfs/folder_block_ops.go

```diff
@@ -532,10 +532,13 @@ func (fbo *folderBlockOps) GetDirBlockForReading(ctx context.Context,
 // assumed that some coordinating goroutine is holding the correct
 // locks, and in that case `lState` must be `nil`.
 //
+// file is used only when reporting errors and sending read
+// notifications, and can be empty.
+//
 // This method also returns whether the block was already dirty.
 func (fbo *folderBlockOps) getFileBlockLocked(ctx context.Context,
 	lState *lockState, kmd KeyMetadata, ptr BlockPointer,
-	file path, rtype blockReqType) (
+	branch BranchName, file path, rtype blockReqType) (
```

Comment:
I did that because of the comment above, that `file` may be empty.  In practice every caller sets a branch, even on an otherwise empty `path`.  But nothing enforces that, so I'll add that as a requirement to the function and drop this param.